builder/frameworks/arduino.py

```
@@ -28,250 +28,635 @@
+        if exists(file_path):
+            safe_delete_file(file_path)
+
+# Initialization
 env = DefaultEnvironment()
 pm = ToolPackageManager()
 platform = env.PioPlatform()
 config = env.GetProjectConfig()
 board = env.BoardConfig()
+
+# Initialize ComponentManager
+component_manager = ComponentManager(env)
```

Ruff (0.11.9)

Undefined name `ComponentManager`

(F821)